app/src/main/java/swati4star/createpdf/activity/FavouritesActivity.java

```diff
@@ -140,4 +152,40 @@ private void onBackPressedState() {
         mSharedPreferences.edit().putBoolean(EXCEL_TO_PDF_KEY, mKeyState[19]).apply();
         mSharedPreferences.edit().putBoolean(ZIP_TO_PDF_KEY, mKeyState[20]).apply();
     }
+
+    private void checkAndAskForStorageDir() {
+        if (Preference.getStringPref(this, STORAGE_LOCATION).isEmpty() || DirectoryUtils.isStorageDirNotExist(this)) {
+            askUserToSelectStorageDir();
+        }
+    }
+
+    private void askUserToSelectStorageDir() {
+        new MaterialAlertDialogBuilder(this).setTitle("Storage folder not found!")
+                .setMessage("Storage directory not found. Please select a folder to save PDF")
+                .setCancelable(false)
+                .setNeutralButton("Choose Folder", (dialog, which) -> {
+                    Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT_TREE);
+                    startActivityForResult(intent, Constants.REQUEST_CODE_FOR_ACTION_OPEN_DOCUMENT_TREE);
+                }).show();
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (data == null || resultCode != RESULT_OK)
+            return;
+        if (requestCode == Constants.REQUEST_CODE_FOR_ACTION_OPEN_DOCUMENT_TREE) {
+            Uri uri = DocumentsContract.buildDocumentUriUsingTree(data.getData(), DocumentsContract.getTreeDocumentId(data.getData()));
+            String storagePath = RealPathUtil.getInstance().getRealPath(this, uri);
+            Preference.setStringPref(this, STORAGE_LOCATION, storagePath);
+            Preference.setStringPref(this, STORAGE_LOCATION_URI, data.getData().toString());
+            DirectoryUtils.getPersistablePermissionOfStorageDir(this, data.getData());
```

Comment:
Can we use "mSharedPreferences" to modify preference?

These lines are also getting repeated everywhere. We don't want code duplicacy. Can it be added as util somewhere?

Reply:
We can use "mSharedPreferences" but modifying preference through "Preference.java" class will be hassle free. It will reduce boilerplate code just - 
Preference.get.......() to get shared preference value
Preference.set........() to put value in shared preference

It will no longer need to define mSharedPreference member variable anymore and no longer need lengthy code to set and get values.

Reply:
Yeah, added it as a util function in commit 1ccbc7f